site/tests/RepoCrawler/WebNotificationConsumerTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Tests\RepoCrawler;
+
+use App\Consumer\WebNotificationConsumer;
+use App\Entity\User;
+use App\Util\Notification;
+use Mgilet\NotificationBundle\Manager\NotificationManager;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Tests for the Web Notification Consumer
+ *
+ * @author Aquib Baig <aquibbaig97@gmail.com>
+ */
+class WebNotificationConsumerTest extends TestCase
+{
+    protected $notification;
+
+    /** @test */
+    public function setUp()
+    {
+        $this->notification = new Notification();
+        $this->notification->setMessage('Your project was added to LibreCores');
+        $this->notification->setSubject('New Project');
+        $this->notification->setType('new_project');
+        $user = new User();
+        $user->setUsername('test');
+        $this->notification->setRecipient($user);
+
+        // Assert if fields are set properly
+        $this->assertEquals('Your project was added to LibreCores', $this->notification->getMessage());
+        $this->assertEquals('New Project', $this->notification->getSubject());
+        $this->assertEquals('new_project', $this->notification->getType());
+        $this->assertEquals('test', $this->notification->getRecipient()->getUsername());
+    }
+
+    /**
+     * @test
+     *
+     * Tests if a notification is added to database
+     */
+    public function test_if_notification_is_added()
+    {
+        $mockLogger = $this->createMock(LoggerInterface::class);
+        $mockNotificationManager = $this->getMockBuilder(NotificationManager::class)
+            ->setMethods(['addNotification'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $webNotificationConsumer = new WebNotificationConsumer($mockLogger, $mockNotificationManager);
+        $webNotificationConsumer->setNotification($this->notification);
```

Comment:
That doesn't test what is really happening. What you should do instead is run the execute() function with a serialized notification object.